ambry-frontend/src/main/java/com/github/ambry/frontend/AmbrySecurityService.java

```diff
@@ -125,6 +128,10 @@ public void postProcessRequest(RestRequest restRequest, Callback<Void> callback)
         exception = new RestServiceException("Too many requests", RestServiceErrorCode.TooManyRequests);
       } else if (storageQuotaService != null && storageQuotaService.shouldThrottle(restRequest)) {
         exception = new RestServiceException("StorageQuotaExceeded", RestServiceErrorCode.TooManyRequests);
+      } else if (quotaManager.getQuotaConfig().requestThrottlingEnabled && quotaManager.shouldThrottle(restRequest)
```

Comment:
The double call to shouldThrottle() looks really weird.  I think QuotaManager.shouldThrottle should be renamed if it doesn't actually tell you if you should throttle.  I would also expect the shouldThrottle method to include the config check internally.

Reply:
If not for config check, we would have to add a check for `quotaManager != null` , which actually shouldn't be null if `requestThrottlingEnabled` config is set to true. Hence I felt this check here is better than null check. But I can change this if thats the preferred approach.
Will rename `QuotaManager`'s `shouldThrottle`.

Reply:
I didn't really follow that.  If quotaManager is null, quotaManager.getQuotaConfig() will throw a NPE.